File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        mp = {}
        for x in nums:
            mp[x] = mp.get(x,0) + 1
        # Now we have constructed our hash map
        # we need to sort by values (frequencies)
        sorted_mp = dict(sorted(mp.items(),key=lambda item : item[1],reverse=True))
        ans = []
        for key in sorted_mp.keys():
            if(k == 0): break
            else:
                ans.append(key)
                k-=1
        return ans

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> mp;
        for(auto x:nums){
            mp[x]++;
        }

        vector<pair<int,int>> arr;
        for(auto x: mp){
            arr.push_back({x.second,x.first});
        }

        sort(arr.rbegin(),arr.rend());

        vector<int> res;
        for(int i=0;i<k;i++){
            res.push_back(arr[i].second);
        }

        return res;
    }
};